src/Native/Windows/gen-buildsys-win.bat

```diff
@@ -10,8 +10,12 @@ if %1=="/?" GOTO :USAGE
 
 setlocal
 set __sourceDir=%~dp0
-:: VS 2015 is the minimum supported toolset
-set __VSString=14 2015
+
+:: Set Visual Studio version from input parameter (build-native.cmd). VS 2015 is the minimum supported toolset
+set __VSString=%2
+set __VSString=%__VSString:"=%
+
+if "%__VSString%"=="" (set __VSString=14 2015)
```

Comment:
This should probably be doing the detection logic for whether we are in a VS2017 Developer Command Prompt, and setting the value to `15 2017` if we are; otherwise falling back to `14 2015`